x-pack/plugins/spaces/server/saved_objects/spaces_saved_objects_client.ts

```diff
@@ -164,19 +166,27 @@ export class SpacesSavedObjectsClient implements SavedObjectsClientContract {
     let namespaces = options.namespaces;
     if (namespaces) {
       const spacesClient = await this.getSpacesClient;
-      const availableSpaces = await spacesClient.getAll('findSavedObjects');
-      if (namespaces.includes('*')) {
-        namespaces = availableSpaces.map((space) => space.id);
-      } else {
-        namespaces = namespaces.filter((namespace) =>
-          availableSpaces.some((space) => space.id === namespace)
-        );
-      }
-      // This forbidden error allows this scenario to be consistent
-      // with the way the SpacesClient behaves when no spaces are authorized
-      // there.
-      if (namespaces.length === 0) {
-        throw this.errors.decorateForbiddenError(new Error());
+      const emptyResponse = SavedObjectsUtils.createEmptyFindResponse<T>(options);
+
+      try {
+        const availableSpaces = await spacesClient.getAll('findSavedObjects');
+        if (namespaces.includes('*')) {
+          namespaces = availableSpaces.map((space) => space.id);
+        } else {
+          namespaces = namespaces.filter((namespace) =>
+            availableSpaces.some((space) => space.id === namespace)
+          );
+        }
+        if (namespaces.length === 0) {
+          // return empty response, since the user is unauthorized in this space (or these spaces), but we don't return forbidden errors for `find` operations
+          return emptyResponse;
```

Comment:
NIT: I would move the `createEmptyFindResponse` call here (and in the catch block) to avoid calling it when unneeded.